valhalla/sif/dynamiccost.h

```diff
@@ -25,6 +25,49 @@
 #include <rapidjson/document.h>
 #include <unordered_map>
 
+// macros aren't great but writing these out for every option is an abomination worse than this macro
```

Comment:
i know macros are generally frowned upon but in this instance it just made so much sense. for every costing option we need to do something very formulaic. in this case we have costing options on which we need to call one of the `set_` methods. the thing we need to set needs to change for every member of costing options but also the type needs to change that we need to pull out of json, we also need to check that the values are in range and if there was no json and there was no preset pbf object we need to use a default.

doing this for one costing option is already quite verbose, and in the end the macro doesnt solve verbosity of what is actually compiled but it makes the costing parsing a hell of a lot more readible and concise. in fact its so very straight forward that i might suggest that we put all costing options parsing into a single function in dynamic cost at some point. pedestrian and bike have a tiny bit of extra logic that needs to be employed for parsing depending on transport type but i think that we could overcome this by having them properly setting their base costing options.

also note there are two flavors of this macro, one for getting an option set and changing that the value is in range using the `ranged_default_t` and the other where you provide just the default and no validity check is done (usually boolean or string values)